frontstage/views/sign_in.py

```diff
@@ -31,70 +31,38 @@ def login():
     if request.method == 'POST' and form.validate():
         username = request.form.get('username')
         password = request.form.get('password')
-
-        # TODO Consider moving this to a helper function.
-        # Lets get a token from the OAuth2 server
-        try:
-            token_url = app.config['ONS_TOKEN']
-
-            data = {
-                'grant_type': 'password',
-                'client_id': app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                'client_secret': app.config['RAS_FRONTSTAGE_CLIENT_SECRET'],
-                'username': username,
-                'password': password,
-            }
-            headers = {
-                'Accept': 'application/json',
-                'Content-Type': 'application/x-www-form-urlencoded;charset=UTF-8',
-            }
-
-            oauth2_response = requests.post(url=token_url, data=data,
-                                            headers=headers, auth=(app.config['RAS_FRONTSTAGE_CLIENT_ID'],
-                                                                   app.config['RAS_FRONTSTAGE_CLIENT_SECRET']))
-            # Check to see that this user has not attempted to login too many times or that they have not forgot to
-            # click on the activate account URL in their email by checking the error message back from the OAuth2 server
-            if oauth2_response.status_code == 401:
-                oauth2Error = json.loads(oauth2_response.text)
-                if 'Unauthorized user credentials' in oauth2Error['detail']:
+        sign_in_data = {
+            "username": username,
+            "password": password
+        }
+        response = api_call('POST', app.config['SIGN_IN_URL'], json=sign_in_data)
+
+        # Handle OAuth2 authentication errors
+        if response.status_code == 502:
+            error_json = json.loads(response.text).get('error')
+            if error_json.get('status_code') == 401:
```

Comment:
Why isn't the `response.status_code == 401`?

Reply:
502 is the correct status code here I think, Bad Gateway indicating an upstream service has returned an invalid response? I can then pass info about that failure in the returned json

Reply:
401 will be used for when Basic Auth etc to the frontstage-api fails

Reply:
I'd disagree. We'd be confusing a microservice with proxy server in that case. 

However orchestrating these changes to return a 401 or a 403 might not be very easy since the pattern has already been introducedl

Reply:
It's not loads of hassle if it does have to be done. This pattern does make things quite clean for the most part other than the cases where we have to return more specific error messages from the origin api's like this one

Reply:
yeah looking through it would be simple to change the api to pass through the status code from the failed service but this would be the 3rd time I've flipped it one way or the other 👀